src/settings_gui.cpp

```diff
@@ -175,6 +175,62 @@ static StringID VolumeMarkFunc(int, int mark, int value)
 	return STR_GAME_OPTIONS_VOLUME_MARK;
 }
 
+static const int FONT_HALF_GAP = _default_font_min_max_gap / 2;
+static const int FONT_SLIDER_MARK_COUNT = _default_font_min_max_gap + 1; // Labeled at every other number. 8 Labeled marks and 7 unlabeled ones.
+
+/**
+ * Called to get the mark string for the font size sliders. Labels every other number starting from the min_*_font_size
+ *
+ * @param mark_count Ignored here but the number of marks on the slider.
+ * @param mark The mark number.
+ * @param value The value of the slider at that mark.
+ * @return A string ID for the string this mark uses. Either raw_int (labelled) or empty string.
+ */
+static StringID FontSizeMarkFunction([[maybe_unused]] int mark_count, int mark, int value)
+{
+	if ((mark & 1) == 1) {
+		return STR_NULL;
+	}
+
+	SetDParam(0, value);
+
+	return STR_JUST_INT;
+}
+
+/**
+ * Calculates the starting mark value for a font size slider. Taking into account the allowed minimum and the need to have
+ * FONT_SLIDER_MARK_COUNT marks. (The max allowed font size value is currently the same for all fonts)
+ *
+ * @param current_size The current size of the font in question.
+ * @param minimum_size The allowed minimum value for this font size.
+ * @return An int value that serves as the minimum value that will be on a slider.
+ */
+static int CalculateFontSliderMin(int current_size, int minimum_size)
+{
+	// This works fine unless the scale is towards the high end of the allowable range:
+	int first_check = std::max(current_size - FONT_HALF_GAP, minimum_size);
+
+	// Adjust for the cases where marks are missing because the top end of the slider has hit the max value:
+	return std::min(first_check, _default_font_sane_max_height_limit - _default_font_min_max_gap);
+}
+
+/**
+ * Calculates the ending mark value for a font size slider. Taking into account the allowed maximum and the need to have
+ * FONT_SLIDER_MARK_COUNT marks. (The max allowed font size value is currently the same for all fonts)
+ *
+ * @param current_size The current size of the font in question.
+ * @param minimum_size The allowed minimum value for this font size.
+ * @return An int value that serves as the maximum value that will be on a slider.
+ */
+static int CalculateFontSliderMax(int current_size, int minimum_size)
+{
+	// This works fine unless the scale is towards the low end of the allowable range:
```

Comment:
More comments with the wrong format. Same note on `first_check` -- maybe something about checking the max height?